documentation/src/main/asciidoc/server_guide/configuration.adoc

```diff
@@ -498,3 +498,36 @@ and editing the module definition to include the jar as resource-root:
 
 ----
 
+==== Custom Marshaller Bridges
+Infinispan provides two marshalling bridges for marshalling client/server requests using the Kryo and Protostuff libraries.
+To utilise either of these marshallers, you simply place the dependency of the marshaller you require in your client
+pom. e.g. for Protostuff:
+
+[source,xml]
+----
+<dependency>
+  <groupId>org.infinispan</groupId>
+  <artifactId>infinispan-marshaller-protostuff</artifactId>
+  <version>{infinispanversion}</version>
+</dependency>
+----
+
+Custom schemas for object marshalling must then be registered with the selected library using the libraries api on the client.
+
+===== Compatibility Mode
+As previously stated, when utilising a custom marshaller in compatibility mode, it is necessary for the marshaller to be on
+the classpath of the server. In order to achieve this with either the Protostuff or Kryo bridges, it is necessary for both
+the bridge class files and the underlying library's runtime dependencies to be on the classpath. To aid with this step,
+we have created a "bundle" jar for each of the bridge implementations which includes all of the runtime class files required
+by the bridge and underlying library.  Therefore, it is only necessary to include this single jar on the server's classpath.
+
+To create a bundle jar:
```

Comment:
This is not user friendly. You can't expect the user to check out the right branch of Infinispan git repo and execute this. The bundle should be an artifact that's uploaded along with other jars to the repository. If they want to access the bundle for a particular version of Infinispan, you can provide the URL to download it from.